lib/pbench/server/cache_manager.py

```diff
@@ -82,6 +84,13 @@ def __str__(self) -> str:
         return f"An error occurred while changing file permissions of {self.tarball}: {self.error}"
 
 
+class TarInfo:
+    def __init__(self, path):
+        self.location = path
+        mtime = os.path.getmtime(path)
+        self.timestamp = datetime.fromtimestamp(mtime).strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
More or less; although I'd prefer to just record a timestamp when we cached an object rather than relying on filesystem attributes. We are, after all, moving away from a filesystem unpack tree and holding on to filesystem attributes here really just complicates things.

Reply:
I don't care whether we maintain the timestamp or whether we rely (if it's feasible) on the underlying storage implementation.  If the storage implementation (whatever it is) is going to maintain a DLU anyway, this saves us from having to write it somewhere reasonably persistent.

But, the real point is, since we're not currently using this value, let's leave it out of the code until we want it...at which point, we can either add it or make use of whatever storage facilities we have at that time.

Reply:
Ok, i will leave the addition to details out for now and work on structure and API access in this PR.

Reply:
So, what kind of info do we wanna return if the path in ` cachemanager.get_info(path)` is a directory or if it is a file?

Reply:
Presumably, we want to return all the information we have about that object, which, at the moment, might be nothing.  🥴

That is, we'll want to return an indication of whether the path references a known object, whether that object is currently in cache, maybe some sort of magic cookie for obtaining its contents (although that might come from a different accessor function based on the path), timestamp (for date last read), and similar...but, since we're not currently maintaining the cache, we don't/won't have the timestamp, or access magic, or any notion of whether it is "in cache" or not; so I think all the function can return is "success" or "not-found"...which isn't much, but it provides the framework for adding all that other stuff once we have it.

As for the difference between files and directories, that depends....  My answer in the preceding paragraph applies to files; presumably there won't be anything to cache for directories, and so most of that other stuff won't be meaningful.  So, what we return for directories (and, presumably, symlinks) will just be navigational information which allows the caller to traverse the tree until they reach a file...if we want to return anything at all for them.

Reply:
Actually, for some reason I thought of that in JSON, but we should probably return a Python object instance, possibly with a class hierarchy from a `CacheObject`, like

```python
class CacheType(Enum):
    FILE = auto()
    DIR = auto()
    SYMLINK = auto()

@dataclass
class ClassObject:
    file: str
    type: CacheType
    path: str

@dataobject
class CacheFile(CacheObject):
    pass

@dataobject
class CacheDirectory(CacheObject):
    children: list[str]
```

Reply:
>
```python
class ClassObject:
```

Presumably you meant `CacheObject`....

Reply:
> I'm not sure what the value of the `"file"` and `"path"` keys are -- doesn't the caller already know that? (Isn't that what it provided to obtain this information?) And, I would only include the `"children"` field if we intend on the caller using this interface in order to walk the file tree; otherwise, I would omit that, too. (Which, currently, leaves us with only `"type"`. upside_down_face)

Yes, we need to support traversing the tree to discover what's there, and the `children` are necessary to support that. I don't see a point in a separate API for that.

The point of `path` is to represent a target for a `cachemanager.get()` call. In a simple example (like right now) it might be a relative filesystem path, as implied by the JSON sample value. When we move off the local filesystem, it might be the actual Redis or S3 URI; or at least a handle the cache manager can translate into the actual location. (Again, we can abstract and hide this, or we can just not bother as this is an internal Pbench API and "they is us".) We might even use `get_info` as a front end both to validate a relative path and to get the `CacheObject` representing it, which might be the parameter to `cachemanager.get` in case we want to propagate more context than just an access "path" string...

Reply:
> The point of `path` is to represent a target for a `cachemanager.get()` call.

This is partly what I was getting at when I was talking about [separating representation and presentation](https://github.com/distributed-system-analysis/pbench/pull/3396#discussion_r1176781192).  I'm expecting that `cachemanager.get()` will receive the same path parameter that `cachemanager.get_info()` takes, that calling the latter is not necessarily a precursor to calling the former (i.e., if the caller already has the path).  I expect that they both would access this information node to obtain the "magic cookie" required to fetch the file contents, and the only question is whether we are willing to expose that magic outside the `cachemanager` implementation or whether we should keep it encapsulated.

That is, the representation of a file in the cache map would include the magic cookie (the S3 URL, or Redis ID, local filesystem path, or whatever), but the presentation returned by `get_info()` might not include it.  (And, in any case, the magic cookie value only exists if the object is in the cache, so it might not even be available from `get_info()`, which implies that `get()` had better have some other means of locating the object.